pkg/apis/softwarecomposition/validation/valdiation_test.go

```diff
@@ -0,0 +1,124 @@
+package validation
+
+import (
+	"testing"
+
+	"github.com/kubescape/storage/pkg/apis/softwarecomposition"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func TestValidateNetworkNeighbors(t *testing.T) {
+	tests := []struct {
+		name             string
+		networkNeighbors softwarecomposition.NetworkNeighbors
+		expectedErrors   field.ErrorList
+	}{
+		{
+			name: "valid",
+			networkNeighbors: softwarecomposition.NetworkNeighbors{
+				Spec: softwarecomposition.NetworkNeighborsSpec{
+					Ingress: []softwarecomposition.NetworkNeighbor{
+						{
+							Identifier: "test",
+							Ports: []softwarecomposition.NetworkPort{
+								{
+									Port:     80,
+									Name:     "UDP-80",
+									Protocol: "UDP",
+								},
+							},
+						},
+					},
+				},
+			},
```

Comment:
issue: validators return an empty error list on success, not nil

So we should expect an empty list. That will also let you greatly simplify your test assertion. We can also improve on naming our test case:
```suggestion
			name: "valid network neighbors should return an empty error list",
			networkNeighbors: softwarecomposition.NetworkNeighbors{
				Spec: softwarecomposition.NetworkNeighborsSpec{
					Ingress: []softwarecomposition.NetworkNeighbor{
						{
							Identifier: "test",
							Ports: []softwarecomposition.NetworkPort{
								{
									Port:     80,
									Name:     "UDP-80",
									Protocol: "UDP",
								},
							},
						},
					},
				},
			},
			expectedErrors: field.ErrorList{},
```